### PR TITLE
upgrade docker image, fix readme, remove deprecated gem-install flags

### PR DESCRIPTION
**Pull request by laur89**
#### Pre-flight Checklist

Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/HEAD/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.

#### Description of Change

- update references to Ferdium 6 -> 7 throughout the documentation/code
- upgrade Dockerfile baseimage to Bookworm (current Debian stable)
- remove deprecated `gem install` options `--no-ri` & `--no-rdoc` that cause installation to fail
- add architecture detection to `pnpm build` in Dockerfile

#### Motivation and Context

Main importance are the Dockerfile baseimage upgrade as currently used image is no longer available. This in turn pulls in newer ruby/gem which causes `gem install` statements to fail due to abovementioned deprecated options.

Basically this change re-enables the ability to build ferdium via provided Dockerfile.

#### Checklist

- [x] My pull request is properly named
- [ ] The changes respect the code style of the project (`pnpm prepare-code`)
- [ ] `pnpm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes

none

### CODE REVIEW

**Review by vraravam — APPROVED** [review]

**Comment by vraravam** [comment]
@all-contributors please add @laur89 for code
